keras_nlp/models/xlm_roberta/xlm_roberta_tasks.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""XLM-RoBERTa task specific models and heads."""
+
+from tensorflow import keras
+
+from keras_nlp.models.roberta import roberta_tasks
+
+
+@keras.utils.register_keras_serializable(package="keras_nlp")
+class XLMRobertaClassifier(roberta_tasks.RobertaClassifier):
+    """XLM-RoBERTa encoder model with a classification head.
+
+    Args:
+        backbone: A `keras_nlp.models.XLMRoberta` instance.
+        num_classes: int. Number of classes to predict.
+        hidden_dim: int. The size of the pooler layer.
+
+    Example usage:
+    ```python
+    # Randomly initialized XLM-RoBERTa encoder
+    model = keras_nlp.models.XLMRoberta(
+        vocabulary_size=50265,
+        num_layers=12,
+        num_heads=12,
+        hidden_dim=768,
+        intermediate_dim=3072,
+        max_sequence_length=12
+    )
+
+    # Call classifier on the inputs.
+    input_data = {
+        "token_ids": tf.random.uniform(
+            shape=(1, 12), dtype=tf.int64, maxval=model.vocabulary_size),
+        "padding_mask": tf.constant(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)),
+    }
+    classifier = keras_nlp.models.XLMRobertaClassifier(
+        backbone=model,
+        num_classes=4,
+    )
+    logits = classifier(input_data)
+    ```
+    """
+
+    pass
```

Comment:
This is kind of an interesting one. Doing a direct subclass will fail as soon as you try to add the equivalent of these lines:
https://github.com/keras-team/keras-nlp/blob/0b8bb378a58fafecfd147ab1118591e7d5716643/keras_nlp/models/bert/bert_tasks.py#L80-L85

At that point, you can't subclass, without at least overriding init, and if you do that you might as well just recreate the whole class at the point. Seems like we would either want to:

 - Not subclass, and copy over the entirety of `RobertaClassifier` here. 
 - Add a private method `_get_backbone_class()` that would could override from here so it uses `XLMRoberta` as the backbone class.

@jbischof any thoughts?

We also don't need to really fix this until we actually add the presets, so we could land as is.

Reply:
My thoughts diverge depending on how common inheritance would be:
* If this is a one-off, I'd prefer a little copying to adding indirection everywhere.
* If there are many models with identical structure, then I'd prefer a little indirection to a lot of copying.

My impression is that this is more of a one-off, correct? And maybe we have a slight preference to copying over lots of linked modeling code? I want models to depend on common modules, not each other.

We should totally keep the backbone inheritance however since that's so clean.

Reply:
Yeah, forking seems reasonable to me. If we do see a large need for this, we can always reverse our decision and consolidate the code later on.

Reply:
Cool!

Reply:
Done!